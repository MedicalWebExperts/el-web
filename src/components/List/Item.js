import React from 'react';
import { string, shape } from 'prop-types';
import theme from '../../styles';

const propTypes = {
  text: string.isRequired,
  styles: shape({}),
};

const defaultProps = {
  styles: {},
};

const Item = ({ text, styles }) => (
  <li style={styles}>
    <style jsx>
      {`
        li {
          padding: 0 0 0 20px;
          position: relative;
          color: ${theme.colors.default};
          font-family: ${theme.font.primary};
          font-size: ${theme.font.size}px;
          line-height: 1.6;
        }
        li:before {
          content: '';
          position: absolute;
          left: 0;
          top: 7px;
          right: auto;
          width: 8px;
          height: 8px;
          border-bottom: solid 1px currentColor;
          border-left: solid 1px currentColor;
          transform: rotate(-135deg);
          color: #4a90e2;
        }
      `}
    </style>
    {text}
  </li>
);

Item.propTypes = propTypes;
Item.defaultProps = defaultProps;
export default Item;
