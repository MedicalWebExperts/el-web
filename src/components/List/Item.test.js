import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';

const defaultProps = {
  text: 'Item test',
};

const customStyles = {
  styles: { padding: 20 },
};

describe('Component Item Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Item {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer
      .create(<Item {...Object.assign({}, defaultProps, customStyles)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
