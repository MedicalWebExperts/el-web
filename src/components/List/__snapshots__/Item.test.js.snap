// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Item Snapshot using custom styles 1`] = `
<li
  className="jsx-1154723179"
  style={
    Object {
      "padding": 20,
    }
  }
>
  Item test
</li>
`;

exports[`Component Item Snapshot without arguments, take the default values 1`] = `
<li
  className="jsx-1154723179"
  style={Object {}}
>
  Item test
</li>
`;
